plugin/src/main/java/org/astropeci/omw/listeners/ItemDeployHandler.java

```diff
@@ -103,69 +105,46 @@ private boolean deployStructure(GameTeam team, Location target, PlayerInteractEv
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 2;
-                height = 2;
-                length = 13;
-                break;
-            case GUARDIAN_SPAWN_EGG:
-                structureName = "guardian";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 3;
-                height = 3;
-                length = 8;
-                break;
-            case GHAST_SPAWN_EGG:
-                structureName = "juggernaut";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 3;
-                height = 3;
-                length = 11;
-                break;
-            case WITCH_SPAWN_EGG:
-                structureName = "shieldbuster";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 3;
-                height = 3;
-                length = 15;
-                break;
-            case OCELOT_SPAWN_EGG:
-                structureName = "lightning";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 5;
-
-                width = 3;
-                height = 2;
-                length = 9;
-                break;
-            default:
-                return false;
+        ConfigurationSection missilesC = plugin.getConfig()
+                .getConfigurationSection("missiles");
+
+        if(missilesC == null)
+            return false;
+
+        ConfigurationSection missileConfig = missilesC.getConfigurationSection(material.name());
+
+        if(missileConfig == null)
+            return false;
+        else{
+            structureName = missileConfig.getString("structureName");
+
+            offsetX = missileConfig.getInt("offsetX");
+            offsetY = missileConfig.getInt("offsetY");
+            offsetZ = missileConfig.getInt("offsetZ");
+
+            width = missileConfig.getInt("width");
+            height = missileConfig.getInt("height");
+            length = missileConfig.getInt("length");
         }
 
-        target.setX(target.getX() + (team == GameTeam.GREEN ? 1 : -1) * offsetX);
+        final var mapConfig = plugin.getConfig()
+                .getConfigurationSection("map");
+
+        if(mapConfig == null)
+            return false;
+
+        final var mapSpawnThreshold = mapConfig.getInt("spawnThreshold");
+
+
+        final var dirFactor = team == GameTeam.GREEN ? -1 : 1;
+
+        target.setX(target.getX() + dirFactor * offsetX);
         target.setY(target.getY() - offsetY);
-        target.setZ(target.getZ() + (team == GameTeam.GREEN ? -1 : 1) * offsetZ);
+        if (Math.signum(target.getZ()) == dirFactor)
+            target.setZ(
+                ((Math.min(Math.abs(target.getZ()) + length + offsetZ, mapSpawnThreshold) - length) * dirFactor));
+        else
+            target.setZ(target.getZ() + dirFactor * offsetZ);
```

Comment:
Missiles being able to spawn directly inside the opponent's base is a feature, not a bug. As someone who regularly runs competitive missile wars tournaments I believe such a change would negatively impact the game in a big way. The project also aims to be as close as possible to the version of the game present on the CubeKrowd server.

Reply:
I actually kind of agree with you here. This is my first ever pull request and I just included this commit by accident, sorry about that. Although maybe this could be kept as an optional game extension? `config.yml` could contain a field `settings` where `spawnMissilesInEnemyBase` would be enabled by default, but could be turned off. That way server owners can decide on wether missiles should be able to spawn inside the enemy base and the default would be the way it works on CubeKrowd.

Reply:
Yeah I'm open to having a config option for it, although maybe instead of having a hard coordinate threshold for where missiles can't be spawned, the anti-griefing code could be altered? This would involve editing `canSpawn` in ItemDeployHandler to treat everything as a friendly block. Might be more natural and also would allow spawning missiles which are technically inside the enemy base but don't overlap anything - such as spawning a missile in the hole created by an explosion.